File: lib/domain/location.dart
import 'dart:convert';

import 'dart:typed_data';

class Location {
  String id;
  String name;
  Uint8List thumbnail;
  double price;
  double rating;
  double longitude;
  double latitude;

  Location(this.id, this.name, this.thumbnail, this.price, this.rating,
      this.longitude, this.latitude);

  Location.empty();

  static fromJson(e) {
    Location l = Location.empty();
    l.id = e['id'];
    l.name = e['name'];
    l.thumbnail = base64.decode(e['thumbnail']);
    l.price = e['price'];
    l.rating = e['rating'];
    l.longitude = e['longitude'];
    l.latitude = e['latitude'];

    return l;
  }
}

File: parkingpp-backend/src/main/java/com/parkingpp/parkingppbackend/model/Location.java
package com.parkingpp.parkingppbackend.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.util.UUID;

@Entity
@Table(name = "location")
public class Location {
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(
            name = "UUID",
            strategy = "org.hibernate.id.UUIDGenerator"
    )
    UUID id;
    String name;
    @Column(length = 1500000)
    String thumbnail;
    double price;
    double rating;
    double longitude;
    double latitude;

    public Location() {

    }

    public Location(@JsonProperty("id") UUID id,
                    @JsonProperty("name") String name,
                    @JsonProperty("thumbnail") String thumbnail,
                    @JsonProperty("price") double price,
                    @JsonProperty("rating") double rating,
                    @JsonProperty("longitude") double longitude,
                    @JsonProperty("latitude") double latitude) {
        this.id = id;
        this.name = name;
        this.thumbnail = thumbnail;
        this.price = price;
        this.rating = rating;
        this.longitude = longitude;
        this.latitude = latitude;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getThumbnail() {
        return thumbnail;
    }

    public void setThumbnail(String thumbnail) {
        this.thumbnail = thumbnail;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public double getRating() {
        return rating;
    }

    public void setRating(double rating) {
        this.rating = rating;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }
}
